catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/RESTEndpoint.java

```diff
@@ -417,7 +420,7 @@ public Response getDocument(@Context UriInfo uriInfo, @Context HttpServletReques
      */
     @GET
     @Path("/sources/{sourceid}/{id}")
-    public Response getDocument(@PathParam("sourceid") String sourceid, @PathParam("id") String id,
+    public Response getDocument(@PathParam("sourceid") String encodedSourceId, @PathParam("id") String encodedId,
```

Comment:
Anytime you're assuming that a JAXRS parameter is encoded, you should also be using `@Encoded` to ensure that it does not get decoded, which will then cause the URLDecoder to give you incorrect results.